### PR TITLE
Added capability support to REXEC

### PR DESCRIPTION
**Pull request by hbs**

### CODE REVIEW

**Review by pi-r-p — CHANGES_REQUESTED** [review]
In the case a script do several REXEC on different endpoints, you need to reload capability between each rexec, and it is hard to keep track of the endpoint URL (you cannot read it anymore in the code).

Why not just read patterns from the capability, building a WebAccessController from this, and keep the explicit endpoint in the signature ? (so, keep the same signature)

`rexec` capability will just override warpscript.rexec.endpoint.patterns configuration, and this will be consistent with other capabilities.

File: warp10/src/main/java/io/warp10/script/ext/rexec/RexecWarpScriptExtension.java [hunk 1]
```diff
@@ -18,20 +18,31 @@
...
 
 public class RexecWarpScriptExtension extends WarpScriptExtension {
-  
+
   private static final Map<String,Object> functions;
-  
+
+  private static final boolean useCapability;
+  public static String CAPABILITY = "rexec";
+  private static String CONF_CAPABILITY = "warpscript.rexec.capability";
```
[the remark this answers is not in the document]
Comment (pi-r-p) [reply warp10/src/main/java/io/warp10/script/ext/rexec/RexecWarpScriptExtension.java:30]: As capabilities are not well known yet, the template may also explicit what it is:
```
# enforce the use of CAPADD with a token that contains the authorized patterns. Generate a read or write token that contains attributes { '.cap:rexec' '.*warp.*' }
# the token capability will override warpscript.rexec.endpoint.patterns
```

File: warp10/src/main/java/io/warp10/script/ext/rexec/REXEC.java [hunk 2]
```diff
@@ -62,67 +63,79 @@ public class REXEC extends NamedWarpScriptFunction implements WarpScriptStackFun
...
   public Object apply(WarpScriptStack stack) throws WarpScriptException {
-    
+
+    String capability = Capabilities.get(stack, RexecWarpScriptExtension.CAPABILITY);
+
+    // Discard capability if we are not using capabilities
+    if (!RexecWarpScriptExtension.useCapability() && null != capability) {
+      capability = null;
+    }
+
```
[the remark this answers is not in the document]
Comment (hbs) [reply warp10/src/main/java/io/warp10/script/ext/rexec/REXEC.java:93]: Enabling capabilities means that you CAN use capabilities, not that they become mandatory, therefore it seems natural to revert to the configured default patterns as configured.

**Comment by hbs** [comment]
PTAL

**Review by pi-r-p — APPROVED** [review]

**Pull request merged**